src/components/measureGroups/SupplementalElements.tsx

```diff
@@ -1,7 +1,381 @@
+          true
+        );
+        updateMeasure(submitMeasure);
+      })
+      .catch(() => {
+        const message = `Error updating measure "${measure.measureName}"`;
+        setErrorMessage(message);
+
+        handleToast(
+          "danger",
+          `Error updating measure "${measure.measureName}"`,
+          true
+        );
+      });
+  };
+
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  // Route handle store will give warning message
+  // if form is dirty and user ties to navigate across SPAs
+  const { updateRouteHandlerState } = routeHandlerStore;
+  const [discardDialogOpen, setDiscardDialogOpen] = useState(false);
+
+  useEffect(() => {
+    updateRouteHandlerState({
+      canTravel: !formik.dirty,
+      pendingRoute: "",
+    });
+  }, [formik.dirty]);
+
+  const handleDefinitionChange = (selectedValues: string[]) => {
+    const oldSelectedSupDataList = cloneDeep(selectedSupplementalData);
+
+    const newList: SupplementalData[] = [];
+    selectedValues.forEach((selectedDefinition) => {
+      newList.push({
+        definition: selectedDefinition,
+        description: getDescriptionByDefinition(
+          selectedDefinition,
+          oldSelectedSupDataList
+        ),
+      });
+    });
+
+    setSelectedSupplementalData(newList);
+    formik.setFieldValue("supplementalData", newList);
+  };
+  const getDescriptionByDefinition = (
+    definition: string,
+    oldSelectedSupDataList: SupplementalData[]
+  ): string => {
+    let description = "";
+    oldSelectedSupDataList.forEach((selected) => {
+      if (definition === selected.definition) {
+        description = selected.description;
+      }
+    });
+    return description;
+  };
+
+  const handleDesciptionChange = (definition, value, index) => {
+    const supData = formik.values.supplementalData;
+    for (let i = 0; i < supData.length; i++) {
+      if (supData[i].definition === definition) {
+        supData[i].description = value;
+      }
+    }
+    formik.setFieldValue("supplementalData", supData);
+
+    const descriptions = supData?.map((data) => {
+      return data.description;
+    });
+    formik.setFieldValue("selectedDescriptions", descriptions);
+  };
+
+  return (
+    <form
+      id="measure-details-form"
+      onSubmit={formik.handleSubmit}
+      data-testid="supplemental-data-form"
+    >
+      <MeasureGroupAlerts {...alertMessage} />
+      <div className="content">
+        <div className="subTitle">
+          <h2>Supplemental Data</h2>
+          <div>
+            <Typography
+              style={{ fontSize: 14, fontWeight: 300, fontFamily: "Rubik" }}
+            >
+              <span style={asterisk}>*</span>
+              Indicates required field
+            </Typography>
+          </div>
+        </div>
+        <div>
+          {allDefinitions && (
```

Comment:
Check is not required, as allDefinitions is initiated with an empty array, so it will be true always.

Reply:
@RohitKandimalla Thanks will remove

Reply:
@RohitKandimalla Done.